common/dbconfig.cpp

```diff
@@ -0,0 +1,103 @@
+// Copyright 2021-present Pinterest. All Rights Reserved.
+//
+// @author Premkumar (prem@pinterest.com)
+//
+
+#include "common/dbconfig.h"
+#include "common/segment_utils.h"
+#include <utility>
+
+#include <gflags/gflags.h>
+#include <glog/logging.h>
+
+DECLARE_int32(replicator_replication_mode);
+
+namespace common {
+
+DBConfigManager* DBConfigManager::get() {
+  static DBConfigManager instance;
+  return &instance;
+}
+
+DBConfigManager::DBConfigManager() : dbConfig_(new DBConfig()) {
+}
+
+bool DBConfigManager::loadJsonStr(const std::string& content) {
+  Json::Value root;
+  Json::Reader reader;
+  if (!reader.parse(content, root) || !root.isObject()) {
+    LOG(ERROR) << "Could not load config";
+    return false;
+  }
+
+  return loadJsonObject(root);
+}
+
+bool DBConfigManager::loadJsonObject(const Json::Value& root) {
+  try {
+    auto newConfig = parseConfig(root);
+    if (newConfig == nullptr) {
```

Comment:
this will never be true. Because parseConfig() always returns a non-nullptr.

How do we want to handle a misformated config?

Reply:
Good Catch . We have changed the code so many times now.  It was supposed to  return nullptr ..  Will add that in.